newsroom/users/users.py

```diff
@@ -19,6 +20,32 @@
 from newsroom.companies.utils import get_company_section_names, get_company_product_ids
 
 
+class UserAuthentication(SessionAuth):
+    def authorized(self, allowed_roles, resource, method):
+        if super().authorized(allowed_roles, resource, method):
+            return True
+
+        if not request.view_args or not request.view_args.get("_id"):
+            # not a request for a specific user, stop
+            return False
+
+        if request.view_args["_id"] == str(get_user_id()):
+            # current user editing current user
+            return True
+
+        current_user = get_user()
+        if not current_user.get("company") or current_user.get("user_type") != UserRole.COMPANY_ADMIN.value:
```

Comment:
Why return `False` if current user has no company?

Reply:
because here the last possible case is it would be company admin for the edited user, but if there is no company then there won't be any link